dashboard/app/serializers/api/v1/pd/enrollment_flat_attendance_serializer.rb

```diff
@@ -1,6 +1,14 @@
 # Write a flat list of workshop attendance by session for the enrollment
 class Api::V1::Pd::EnrollmentFlatAttendanceSerializer < ActiveModel::Serializer
-  attributes :first_name, :last_name, :email
+  attributes :first_name, :last_name, :email, :district_name, :school, :role, :grades_teaching
+
+  def district_name
+    object.try(:school_info).try(:school_district).try(:name)
```

Comment:
For my learning, do you have preferences on using try vs. &. vs. etc? Was just looking through this article:

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

Reply:
This very short answer captures my (surely incomplete) understanding about their differences https://stackoverflow.com/a/45825477 :)

Reply:
I realize that we have been inconsistent in using `try` and `&.`. To be consistent, we should be clear about 2 things
1. What object kinds can be serialized by a specific serializer? In this case, if the input object is guaranteed to be an Enrollment or nil, using `&.` should be enough. If the object could be of any kind, then we should use `try`.
2. How to handle failure when the input object does not have expected fields? We can either fail silently (return nil) or loudly (raise exception).

I will change this code to use `&.` with the assumption that an input object must be an Enrollment. The code will raise exception otherwise.